topi/python/topi/transform.py

```diff
@@ -269,3 +269,22 @@ def matmul(a, b, transp_a=False, transp_b=False):
     A Tensor whose op member is the matmul operation
     """
     return cpp.matmul(a, b, transp_a, transp_b)
+
+
+def tensordot(a, b, ndims_to_reduce):
+    """A generalization of matrix multiplication to tensor.
+
+    `Res[i_1, ... , j_1, ...] = Sum_{k_1, ...} A[i_1 ..., k_1, ...]*B[k_1, ..., j_1, ...]`
+    The number of `k` variables is ndims_to_reduce.
+
+    Parameters
+    ----------
+    a : The tensor A
+    b : The tensor B
+    ndims_to_reduce : The number of dimensions to reduce over
```

Comment:
Could you change to use `axis` as with [numpy's tensordot](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.tensordot.html)?
I think It is better to make it compatible with numpy if possible.